examples/miral-shell/shell_main.cpp

```diff
@@ -135,6 +138,7 @@ int main(int argc, char const* argv[])
 
     return runner.run_with(
         {
+            CustomDecorations{UserDecorationManagerExample::get_adapter()},
```

Comment:
Is the adaptor created once on program startup and used for all decorations? Or might it need to maintain state associated with each window (and therefore be created once per window?)

Reply:
> Is the adaptor created once on program startup and used for all decorations? 

Yep.

> Or might it need to maintain state associated with each window (and therefore be created once per window?)
As it stands with `BasicDecoration`, the manager keeps track of decorations, and each decoration has it's own window-specific state.